pandas/core/apply.py

```diff
@@ -442,6 +448,79 @@ def agg_list_like(self) -> DataFrame | Series:
             )
             return concatenated.reindex(full_ordered_index, copy=False)
 
+    def hom_list_single_arg(
+        self, method: str, a: AggFuncTypeBase, result_dim: int | None
+    ) -> tuple[int | None, AggFuncTypeBase | None, DataFrame | Series | None]:
+        result = None
+        if isinstance(a, (tuple, list)):
+            # Handle (name, value) pairs
+            name, a = a
+        else:
+            name = com.get_callable_name(a) or a
+        try:
+            result = getattr(self.obj, method)(a)
+        except (TypeError, DataError):
+            warnings.warn(
+                f"{name} did not aggregate successfully. If any error is "
```

Comment:
why do we want to warn here? (e.g. new api can we just raise)

Reply:
Right - new API will just raise. Starting out, I'd like to maintain as much consistency between `api.use_hom=True` and `api.use_hom=False` as possible, only making changes that would be hard/impossible to properly deprecate. Since this is already deprecated and will be changed in 2.0, I don't see a need to do it here.